test/graph_construction_tests.jl

```diff
@@ -1715,3 +1715,64 @@ end
     @test length(collect(filter(as_variable(:in), model))) == 3
     @test length(collect(filter(as_variable(:out), model))) == 2
 end
+
+@testitem "Comparing variables throws warning" begin
+    import GraphPPL: create_model, getorcreate!
+
+    include("testutils.jl")
+    @model function test_model(y)
+        x ~ Normal(0.0, 1.0)
+        if x == 0
+            z ~ Normal(0.0, 1.0)
+        else
+            z ~ Normal(1.0, 1.0)
+        end
+    end
+
+    @test_logs (
+        :warn,
+        "Comparing Factor Graph variable (x) with a value. This is not possible as the value of x is not known at model construction time."
+    ) create_model(test_model(y = 1))
+
+    @model function test_model(y)
+        x ~ Normal(0.0, 1.0)
+        if x > 0
+            z ~ Normal(0.0, 1.0)
+        else
+            z ~ Normal(1.0, 1.0)
+        end
+    end
+
+    @test_logs (
+        :warn,
+        "Comparing Factor Graph variable (x) with a value. This is not possible as the value of x is not known at model construction time."
+    ) create_model(test_model(y = 1))
+
+    @model function test_model(y)
+        x ~ Normal(0.0, 1.0)
+        if x < 0
+            z ~ Normal(0.0, 1.0)
+        else
+            z ~ Normal(1.0, 1.0)
+        end
+    end
+
+    @test_logs (
```

Comment:
Can we add a test where `x` also uses the index. I would expect the warning to be something like `Comparing Factor Graph variable (x[1, 2]) with a value`

Reply:
Also why in the first sentence it is in the brackets and in the second it is not. I would replace the brackets with backticks and add backticks in the second sentence.

Reply:
ping @wouterwln